main/provider/index.ts

```diff
@@ -604,14 +604,12 @@ export class Provider extends EventEmitter {
       if (!Number.isInteger(chainId)) throw new Error('Invalid chain id')
 
       // Check if chain exists 
-      const exists = Boolean(store('main.networks', 'ethereum', chainId))
+      const exists = Boolean(store('main.networks.ethereum', chainId))
       if (exists === false) throw new Error('Chain does not exist')
 
-      const originId = payload._origin
-      const origin = storeApi.getOrigin(originId)
-
+      const origin = getPayloadOrigin(payload)
       if (origin.chain.id !== chainId) {
-        store.switchOriginChain(originId, chainId, origin.chain.type)
+        store.switchOriginChain(origin.chain.id, chainId, origin.chain.type)
```

Comment:
this is supposed to be the origin id not the chain id

Reply:
duh.  my brain must not be functional today